eng/build.yml

```diff
@@ -69,36 +69,42 @@ parameters:
   default: false
 
 jobs:
-- template: common/templates/job/job.yml
+- template: common/templates-official/job/job.yml
   parameters:
     # Set up the name of the job.
     ${{ if parameters.pgoInstrument }}:
       name: PGO_${{ parameters.agentOs }}_${{ parameters.jobName }}
     ${{ if not(parameters.pgoInstrument) }}:
       name: ${{ parameters.agentOs }}_${{ parameters.jobName }}
-    
-    ## Set up the pool/machine info to be used based on the Agent OS
+
+    # Set up the pool/machine info to be used based on the Agent OS
     ${{ if eq(parameters.agentOs, 'Windows_NT') }}:
       enableMicrobuild: true
       pool:
         ${{ if eq(variables['System.TeamProject'], 'public') }}:
           name: $(DncEngPublicBuildPool)
-          demands: ImageOverride -equals windows.vs2022.amd64.open
+          image: 1es-windows-2022-open
+          os: windows
         ${{ if eq(variables['System.TeamProject'], 'internal') }}:
           name: $(DncEngInternalBuildPool)
-          demands: ImageOverride -equals windows.vs2022.amd64
+          image: 1es-windows-2022
+          os: windows
     ${{ if eq(parameters.agentOs, 'Linux') }}:
       pool:
         ${{ if eq(variables['System.TeamProject'], 'public') }}:
           name: $(DncEngPublicBuildPool)
-          demands: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
+          image: 1es-ubuntu-2004-open
```

Comment:
2204?

(I think 2004 would be ok, but it's different than the 2204 below, and I think they should be the same.)

Reply:
2204 doesn't exist for a public agent currently. You can see the list of public agents here: https://helix.dot.net/#1ESHostedPoolImagesWestUS-rg-Public

Technically, I don't need to change this at all since this file isn't used by the public PR pipeline currently. However, I might change my mind on that after starting to work on the other branches. I'd make a new PR if I simplify things.